#Project 2, Part 1

**Name:** Thomas Greenacre

**Date: **7/6/2022

In [8]:
#import libraries
import pandas as pd
import numpy as np

In [1]:
#mount google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


First Choice: Dataset 1

In [4]:
# import dataset 1 and assign to dataframe, show first 5 rows
filename = '/content/drive/MyDrive/Coding Dojo/Week 7/Project 2 - Part 1/car_price_prediction[1].csv'
df_1 = pd.read_csv(filename)
df_1.head()

ID  Price  Levy Manufacturer    Model  Prod. year   Category  \
0  45654403  13328  1399        LEXUS   RX 450        2010       Jeep   
1  44731507  16621  1018    CHEVROLET  Equinox        2011       Jeep   
2  45774419   8467     -        HONDA      FIT        2006  Hatchback   
3  45769185   3607   862         FORD   Escape        2011       Jeep   
4  45809263  11726   446        HONDA      FIT        2014  Hatchback   

  Leather interior Fuel type Engine volume    Mileage  Cylinders  \
0              Yes    Hybrid           3.5  186005 km        6.0   
1               No    Petrol             3  192000 km        6.0   
2               No    Petrol           1.3  200000 km        4.0   
3              Yes    Hybrid           2.5  168966 km        4.0   
4              Yes    Petrol           1.3   91901 km        4.0   

  Gear box type Drive wheels   Doors             Wheel   Color  Airbags  
0     Automatic          4x4  04-May        Left wheel  Silver       12  
1     Tiptronic          4x4  04-May        Left wheel   Black        8  
2      Variator        Front  04-May  Right-hand drive   Black        2  
3     Automatic          4x4  04-May        Left wheel   White        0  
4     Automatic        Front  04-May        Left wheel  Silver        4

In [11]:
#determine # of rows and features (columns)
np.shape(df_1)

(19237, 18)

In [12]:
# import dataset 2 and assign to dataframe, show first 5 rows
filename = '/content/drive/MyDrive/Coding Dojo/Week 7/Project 2 - Part 1/VehicleInsuranceData[1].csv'
df_2 = pd.read_csv(filename)
df_2.head()

Unnamed: 0           clv Response  Coverage Education EmploymentStatus  \
0           1   2763.519279       No     Basic  Bachelor         Employed   
1           2   6979.535903       No  Extended  Bachelor       Unemployed   
2           3  12887.431650       No   Premium  Bachelor         Employed   
3           4   7645.861827       No     Basic  Bachelor       Unemployed   
4           5   2813.692575       No     Basic  Bachelor         Employed   

  Gender  Income Location.Code Marital.Status  ...  \
0      F   56274      Suburban        Married  ...   
1      F       0      Suburban         Single  ...   
2      F   48767      Suburban        Married  ...   
3      M       0      Suburban        Married  ...   
4      M   43836         Rural         Single  ...   

   Months.Since.Policy.Inception  Number.of.Open.Complaints  \
0                              5                          0   
1                             42                          0   
2                             38                          0   
3                             65                          0   
4                             44                          0   

   Number.of.Policies     Policy.Type        Policy Renew.Offer.Type  \
0                   1  Corporate Auto  Corporate L3           Offer1   
1                   8   Personal Auto   Personal L3           Offer3   
2                   2   Personal Auto   Personal L3           Offer1   
3                   7  Corporate Auto  Corporate L2           Offer1   
4                   1   Personal Auto   Personal L1           Offer1   

  Sales.Channel Total.Claim.Amount  Vehicle.Class  Vehicle.Size  
0         Agent         384.811147   Two-Door Car       Medsize  
1         Agent        1131.464935  Four-Door Car       Medsize  
2         Agent         566.472247   Two-Door Car       Medsize  
3   Call Center         529.881344            SUV       Medsize  
4         Agent         138.130879  Four-Door Car       Medsize  

[5 rows x 22 columns]

In [13]:
#inspect features, data tpes, null counts, etc.
df_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8630 entries, 0 to 8629
Data columns (total 22 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Unnamed: 0                     8630 non-null   int64  
 1   clv                            8630 non-null   float64
 2   Response                       8630 non-null   object 
 3   Coverage                       8630 non-null   object 
 4   Education                      8630 non-null   object 
 5   EmploymentStatus               8630 non-null   object 
 6   Gender                         8630 non-null   object 
 7   Income                         8630 non-null   int64  
 8   Location.Code                  8630 non-null   object 
 9   Marital.Status                 8630 non-null   object 
 10  Monthly.Premium.Auto           8630 non-null   int64  
 11  Months.Since.Last.Claim        8630 non-null   int64  
 12  Months.Since.Policy.Inception  8630 non-null   i

In [15]:
#determine # of rows and features (columns)
np.shape(df_2)

(8630, 22)

**First Choice: Dataset 1**

1) Source of data: https://www.kaggle.com/datasets/deepcontractor/car-price-prediction-challenge 

2) Brief description of data: this dataset has information (data) for various car makes & models. Features included in the data set are Manufacturer, Model, Category, Fuel Type, color, etc.

3) What is the target? Price (car price)

4) Is this a classification or regression problem?: Regression

5) How many features?: 18

6) How many rows of data?: 19,237

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
*   determining which features to keep/drop based on if they factor into the target
*   cleaning data, removing dups, imputing for missing values, missing data, etc.



---



**Second choice: dataset 2**

1) Source of data: https://www.kaggle.com/datasets/junglisher/vehicle-insurance-data

2) Brief description of data: this dataset has information for various individual insurance policies. Features included in the data set are Education, Marital Status, Income, Gender, etc. 

3) What is the target?: Monthly.Premium.Auto

4) Is this a classification or regression problem?: Regression

5) How many features?: 22

6) How many rows of data?: 8630

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
*   determining which features to keep/drop based on if they factor into the target
*   cleaning data, removing dups, imputing for missing values, missing data, etc.